sdk/metric/controller/basic/controller.go

```diff
@@ -196,12 +196,10 @@ func (c *Controller) collect(ctx context.Context) error {
 	if c.exporter == nil {
 		return nil
 	}
+
 	// Note: this is not subject to collectTimeout.  This blocks the next
 	// collection despite collectTimeout because it holds a lock.
-	if err := c.export(ctx); err != nil {
-		return err
-	}
-	return nil
```

Comment:
I'm not sure if this structure was intentional, however as the linter complained and the change does not affect the behaviour I applied it.